### PR TITLE
Update libcoap to include support for DTLS Connection IDs

### PR DESCRIPTION
**Pull request by hasheddan**
Updates libcoap to 4abe0ea to pick up DTLS Connection ID support, which was
added in https://github.com/obgm/libcoap/pull/1153.

Signed-off-by: Daniel Mangum <georgedanielmangum@gmail.com>

### CODE REVIEW

**Comment by hasheddan** [comment]
Required update in esp-idf https://github.com/espressif/esp-idf/pull/12177

**Review by hasheddan — COMMENTED** [review]
Bumping into a small issue with Zephyr build of `libcoap` as described here: https://github.com/obgm/libcoap/issues/1217

**Review by hasheddan — COMMENTED** [review]
This is currently now only blocked on https://github.com/espressif/esp-idf/pull/12177, which we could work around for the time-being by not enabling `CONFIG_MBEDTLS_SSL_DTLS_CONNECTION_ID`.

**Review by hasheddan — COMMENTED** [review]
This is now blocked on https://github.com/obgm/libcoap/issues/1225

**Review by sam-golioth — APPROVED** [review]

File: src/golioth_coap_client.c [hunk 1]
```diff
@@ -966,6 +966,9 @@ golioth_client_t golioth_client_create(const golioth_client_config_t* config) {
         time_t t;
         srand(time(&t));
 
+        // Initialize libcoap prior to any coap_* function calls.
+        coap_startup();
```
[the remark this answers is not in the document]
Comment (hasheddan) [reply src/golioth_coap_client.c:970]: Ah good catch -- updated!

**Pull request merged**